backend.native/tests/interop/objc/tests/workerAutoreleasePool.kt

```diff
@@ -0,0 +1,153 @@
+import kotlin.native.concurrent.*
+import kotlinx.cinterop.*
+import kotlin.test.*
+import objcTests.*
+
+@Test
+fun testExecute1() =
+        test(Execute, InMain, Park)
+
+@Test
+fun testExecute2() =
+        test(Execute, InMain, ProcessQueue)
+
+@Test
+fun testExecute3() =
+        test(Execute, InWorker, Park)
+
+@Test
+fun testExecute4() =
+        test(Execute, InWorker, ProcessQueue)
+
+@Test
+fun testExecute5() =
+        test(Execute, InMainToWorker, NoYield)
+
+@Test
+fun testExecuteAfter1() =
+        test(ExecuteAfter, InMain, Park)
+
+@Test
+fun testExecuteAfter2() =
+        test(ExecuteAfter, InMain, ProcessQueue)
+
+@Test
+fun testExecuteAfter3() =
+        test(ExecuteAfter, InWorker, Park)
+
+@Test
+fun testExecuteAfter4() =
+        test(ExecuteAfter, InWorker, ProcessQueue)
+
+@Test
+fun testExecuteAfter5() =
+        test(ExecuteAfter, InMainToWorker, NoYield)
+
+private fun <Job> test(method: ExecuteMethod<Job>, context: Context, yieldMethod: Yield) {
+    context.withWorker { worker ->
+        fun execute(block: () -> Unit) {
+            val job = method.submit(worker, block)
+            yieldMethod.yield()
+            method.wait(job)
+        }
+
+        val deallocated = CreateAutoreleaseDeallocated()
+
+        execute {
+            CreateAutorelease.createAutorelease(deallocated)
+            // Object is still in autorelease pool:
+            assertFalse(deallocated.value)
+        }
+
+        // autorelease pool is processed after the job is finished, so the object should be deallocated;
+        // Checking in a job to make sure previous job is completely processed:
+        execute {
+            assertTrue(deallocated.value)
+        }
+    }
+}
+
+interface ExecuteMethod<Job> {
```

Comment:
According to usage, maybe `Result` is a better name than `Job`

Reply:
`Result` is kinda ambiguous here. Maybe `Future`? Since it might clash with `Future` class, I'd go with `F`. What do you think?

Reply:
`F` is fine. `JobFuture` is another alternative.

Reply:
Done.